lib/std/testing.zig

```diff
@@ -374,6 +374,43 @@ pub const TmpDir = struct {
         self.parent_dir.close();
         self.* = undefined;
     }
+
+    /// Writes program string as zig file into tmp directory
+    /// Caller owns memory
+    ///
+    /// ```
+    /// const progstr = "pub fn main() void {}\n";
+    /// var it = try std.process.argsWithAllocator(std.testing.allocator);
+    /// defer it.deinit(); // no-op unless WASI or Windows
+    /// const testargs = try std.testing.getTestArgs(&it);
+    /// var tmp = std.testing.tmpDir(.{ .no_follow = true }); // ie zig-cache/tmp/8DLgoSEqz593PAEE
+    /// defer tmp.cleanup();
+    /// const zigfile_path = try tmp.writeZigFile(std.testing.allocator, progstr, "bruh");
+    /// defer std.testing.allocator.free(zigfile_path);
+    /// const binary = zigfile_path[0 .. zigfile_path.len - 4]; // '.zig' is 4 characters
+    /// try std.testing.buildExe(testargs.zigexec, zigfile_path, binary);
+    /// ```
```

Comment:
this belongs as a doctest. if you are running into problems with that, then do one of these things:
 * fix them as a prerequisite to this PR
 * work around the issue by using `test "foo" {` syntax instead of `test foo {` syntax.

Reply:
workaround was used.